lib/destroyed_at.rb

```diff
@@ -39,12 +39,25 @@ def destroyed(time = nil)
 
   # Set an object's destroyed_at time.
   def destroy(timestamp = nil)
-    timestamp ||= @marked_for_destruction_at || current_time_from_proper_timezone
-    raw_write_attribute(:destroyed_at, timestamp)
-    run_callbacks(:destroy) do
-      destroy_associations
-      self.class.unscoped.where(self.class.primary_key => id).update_all(destroyed_at: timestamp)
-      @destroyed = true
+    transaction do
+      timestamp ||= @marked_for_destruction_at || current_time_from_proper_timezone
+      raw_write_attribute(:destroyed_at, timestamp)
+
+      run_callbacks(:destroy) do
+        destroy_associations
+        self.class.unscoped.where(self.class.primary_key => id).update_all(destroyed_at: timestamp)
+        @destroyed = true
+
+        next unless ActiveRecord::VERSION::STRING >= '4.2'
```

Comment:
@marktran Thanks for the PR! Could you clarify the use of `next if` and `next unless` within these blocks rather than simply omitting the `next` and relying on `if` and `unless` conditionals. I would think that this could be written as follow, but I may be missing something:

```rb
       if ActiveRecord::VERSION::STRING >= '4.2'
        each_counter_cached_associations do |association|
          foreign_key = association.reflection.foreign_key.to_sym
          unless destroyed_by_association && destroyed_by_association.foreign_key.to_sym == foreign_key
            association.decrement_counters if send(association.reflection.name)
          end
        end
      ...
```